src/main/java/com/google/cloud/tools/maven/Deploy.java

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools.maven;
+
+import com.google.cloud.tools.Action;
+import com.google.cloud.tools.StageGenericJavaAction;
+import com.google.cloud.tools.maven.configs.DeployConfig;
+import com.google.cloud.tools.maven.configs.StageConfig;
+import com.google.cloud.tools.StageAction;
+import com.google.common.base.Strings;
+import com.google.cloud.tools.DeployAction;
+import com.google.cloud.tools.DeployAction.AppType;
+import com.google.cloud.tools.InvalidDirectoryException;
+import com.google.cloud.tools.InvalidFlagException;
+import com.google.cloud.tools.Option;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.Execute;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.codehaus.plexus.util.FileUtils;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.repository.RemoteRepository;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Deploys an app to the server.
+ */
+@Mojo(name = "deploy")
+@Execute(phase = LifecyclePhase.PACKAGE)
```

Comment:
The package phase is when the "target" directory is created with the generated JAR/WAR/etc.

Do you have an alternative suggestion?